### PR TITLE
Clean up `_add_swift_host_library_single`

### PR DESCRIPTION
**Pull request by compnerd**
Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

Resolves SR-NNNN.

### CODE REVIEW

**Comment by compnerd** [comment]
CC: @xiaobai @gottesmm @drexin

**Comment by compnerd** [comment]
@karwa hopefully this explains somewhat the reason for the split - this allows us to nearly remove all the options

**Comment by compnerd** [comment]
@swift-ci please clean test

**Comment by compnerd** [comment]
@swift-ci please test Windows platform

File: cmake/modules/AddSwift.cmake [hunk 1]
```diff
@@ -674,26 +674,12 @@ endfunction()
...
 #     [ARCHITECTURE architecture]
-#     [DEPENDS dep1 ...]
-#     [LINK_LIBRARIES dep1 ...]
-#     [FRAMEWORK_DEPENDS dep1 ...]
-#     [FRAMEWORK_DEPENDS_WEAK dep1 ...]
 #     [LLVM_LINK_COMPONENTS comp1 ...]
-#     [C_COMPILE_FLAGS flag1...]
-#     [SWIFT_COMPILE_FLAGS flag1...]
-#     [LINK_FLAGS flag1...]
-#     [FILE_DEPENDS target1 ...]
```
Comment (drexin) [inline cmake/modules/AddSwift.cmake]: `add_swift_host_library` still defines `FILE_DEPENDS` as one of the parameters, but doesn't use it anywhere, so I'm wondering if it should be removed there as well.

Comment (compnerd) [reply cmake/modules/AddSwift.cmake]: Hmm, perhaps I missed it?  It should be removed in the final view of things.  The individual steps are meant to be easier to digest.

**Comment by compnerd** [comment]
@swift-ci please clean test

**Comment by compnerd** [comment]
@swift-ci please test Windows platform

**Comment by compnerd** [comment]
@swift-ci please test Windows platform

**Comment by compnerd** [comment]
@swift-ci please clean test Windows platform

**Comment by compnerd** [comment]
@swift-ci please test macOS platform

File: cmake/modules/AddSwift.cmake (hunk 1 above)
Comment (compnerd) [reply cmake/modules/AddSwift.cmake]: Hmm, it hasn't supported that for a while, it did however document it as supporting it before this set of changes.  Fake news! :-p

**Comment by compnerd** [comment]
@swift-ci please clean test macOS platform

**Comment by compnerd** [comment]
@swift-ci please test Windows platform

**Comment by compnerd** [comment]
@swift-ci please test Windows platform

**Comment by compnerd** [comment]
@swift-ci please clean test

**Comment by compnerd** [comment]
@swift-ci please clean test

**Comment by compnerd** [comment]
@swift-ci please test Windows platform

**Comment by compnerd** [comment]
@swift-ci please test Windows platform

**Comment by compnerd** [comment]
@swift-ci please clean test

**Comment by compnerd** [comment]
@swift-ci please clean test

**Comment by compnerd** [comment]
@swift-ci please clean test

**Comment by compnerd** [comment]
@swift-ci please clean test macOS platform

**Comment by compnerd** [comment]
@swift-ci please test Linux platform

**Comment by compnerd** [comment]
@swift-ci please clean test

**Comment by compnerd** [comment]
@swift-ci please test Windows platform

**Comment by compnerd** [comment]
@swift-ci please test Windows platform

**Pull request merged**